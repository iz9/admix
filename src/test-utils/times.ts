export function times(amount: number, cb: (...args: any) => any) {
  let i = amount
  while (i > 0) {
    !void cb() && --i
  }
}
