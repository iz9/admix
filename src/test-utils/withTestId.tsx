import React from 'react'

interface WithTestID {
  testId: string
}

export function withTestId<P>(Component: React.ComponentType<P>) {
  const WithTestId: React.FC<P & WithTestID> = props => {
    const { testId, ...rest } = props
    return <Component data-testid={testId} {...(rest as P)} />
  }
  return WithTestId
}
