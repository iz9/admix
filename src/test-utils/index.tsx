import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { withTestId } from './withTestId'
import { ThemeProvider } from '../ui-lib/theme'
import { hexToRGB } from './hexToRgb'

const AllTheProviders: React.FC = ({ children }) => (
  <ThemeProvider>{children as React.ReactChild}</ThemeProvider>
)

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render, withTestId, hexToRGB }
