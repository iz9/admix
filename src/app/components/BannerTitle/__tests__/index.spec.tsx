import React from 'react'
import { withTestId, render, screen, cleanup } from 'test-utils'

import { BannerTitle } from '..'

describe('<BannerTitle  />', () => {
  afterEach(() => {
    cleanup()
  })

  it('should match snapshot', () => {
    const loadingIndicator = render(
      <BannerTitle color={'accentPink'}>Banner text</BannerTitle>,
    )
    expect(loadingIndicator.container.firstChild).toMatchSnapshot()
  })

  it('should render children text uppercase', () => {
    const text = 'text'
    const testID = 'testID'
    const Cmp = withTestId(BannerTitle)
    render(
      <Cmp testId={testID} color={'black'}>
        {text}
      </Cmp>,
    )
    expect(screen.getByTestId(testID).innerHTML).toBe(text.toUpperCase())
  })
})
