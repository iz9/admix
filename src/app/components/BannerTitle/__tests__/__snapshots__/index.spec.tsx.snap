// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerTitle  /> should match snapshot 1`] = `
<span
  class="css-1hbbhby-Text"
>
  BANNER TEXT
</span>
`;
